tracer.go

```diff
@@ -210,3 +210,16 @@ func (t *tracerImpl) Join(operationName string, format interface{}, carrier inte
 	}
 	return nil, opentracing.ErrUnsupportedFormat
 }
+
+// GetGlobalTracerOptions attempts to get the basictracer.Options
+// struct provided to the global Tracer.  Returns the options and true
+// if the global Tracer is a basictracer implementation; returns
+// false otherwise.
+func GetGlobalTracerOptions() (Options, bool) {
```

Comment:
@yurishkuro In our case, for instance, we want to allow the programmer/user to interact with our actual *tracing system* from their application code (not just describe the instrumentation)... e.g., request/trigger trace assembly for an arbitrary Span available within the process.

FWIW, we don't *need* to be able to access the `Recorder` (and thus our basictracer impl, basically)... we could ask the programmer to keep their own reference to our tracing system on hand. But it would be more convenient if that user only had to pass around a single (Tracer) object.

Also, just to clarify: none of the casting here affects the trace structure per se... it's all about getting access to things like span guids/etc, then using them to trigger effects in other systems. (If we were going through this trouble just to modify the trace structure itself I would consider this sort of downcasting very non-OT-ish)

Okay, on balance: I think there is good reason to expose some sort of read API on basictracer Spans. Though we could try leaving the Recorder/Options out of this change and see how far we can get by asking our own users to maintain a separate reference to our tracing system (independent of the `Tracer` which is just used to add instrumentation).